app/services/data/export/send-to-s3-bucket.service.js

```diff
@@ -5,40 +5,224 @@
+            httpAgent: new HttpProxyAgent({ proxy: requestConfig.httpProxy })
+          }
+        : {}),
+      connectionTimeout: 10000
+    })
+  }
 }
 
 /**
- * Uploads a file to an Amazon S3 bucket using the given parameters
+ * Returns the upload type based on its size to the specified S3 bucket
+ *
+ * AWS S3 bucket have conditions on using multi-part uploads. One of these conditions is the file has to be over 5 MB.
+ * Anything under 5MB must be a single upload. By isolating the service in its own file, we can create a stub for
+ * testing upload types with the "send-to-S3-bucket" service.
+ * {@link https://docs.aws.amazon.com/AmazonS3/latest/userguide/qfacts.html S3 multipart upload limits}
+ *
+ * @param {Buffer} buffer The file content as a buffer object
  *
- * @param {Object} params The parameters to use when uploading the file
+ * @returns {Boolean} True if the buffer is smaller than 5 MB else false
  */
-async function _uploadToBucket (params) {
-  const s3Client = new S3Client()
-  const command = new PutObjectCommand(params)
+function _singleUpload (buffer) {
+  const FIVE_MEGA_BYTES = 5242880
 
-  await s3Client.send(command)
+  return buffer.length <= FIVE_MEGA_BYTES
+}
+
+/**
+ * Uploads a file in multiple parts to the specified S3 bucket with the provided key
+ *
+ * @param {Buffer} buffer The file content as a buffer object
+ * @param {String} bucketName Name of the S3 bucket to upload the file to
+ * @param {String} key The path under which the file will be stored in the bucket
+ */
+async function _uploadMultipartFile (bucketName, key, buffer, customConfig) {
+  const s3Client = new S3Client(customConfig)
+  let uploadId
+
+  try {
+    const uploadId = await _createMultipartUpload(s3Client, bucketName, key)
+
+    const uploadResults = await _uploadPartCommand(s3Client, bucketName, key, buffer, uploadId)
+
+    await _completeMultipartUploadCommand(s3Client, bucketName, key, uploadId, uploadResults)
+  } catch (error) {
+    global.GlobalNotifier.omfg('Send to S3 errored', error)
+
+    if (uploadId) {
+      const abortCommand = new AbortMultipartUploadCommand({
+        Bucket: bucketName,
+        Key: key,
+        UploadId: uploadId
+      })
+      await S3Client.send(abortCommand)
+    }
+  }
+}
+
+/**
+ * Uploads a buffer as parts in a multipart upload to an Amazon s3 bucket
+ *
+ * @param {*} s3Client The AWS S3 Client instance used to interact with the service
+ * @param {String} bucketName The name of the bucket where the object will be stored
+ * @param {String} key The key (path) of the object within the bucket
+ * @param {Buffer} buffer The buffer containing the data to be uploaded in parts
+ * @param {String} uploadId The uploadId associated with the created multipart upload
+ *
+ * @returns An array of responses from the upload promises for each part.
+ */
+async function _uploadPartCommand (s3Client, bucketName, key, buffer, uploadId) {
```

Comment:
Apologies, I've got a few comments going on. But I don't think the name of this method corresponds to what it is doing.

My understanding is this is responsible for uploading _all_ the parts, not just a single one? If so it might help with an additional thought; this function is doing too much.

I would suggest renaming and splitting out the work. For example, this becomes `_splitIntoPartsAndUpload()`. Then everything in the `for()` loop gets moved to a function called `_uploadPart()`. 🤷